new/supermarket_namespace_governance.md

```diff
@@ -0,0 +1,167 @@
+---
+RFC: unassigned
+Author: Sean OMeara <sean@chef.io>
+Status: Draft
+Type: <Standards Track, Informational, Process>
+<Replaces: RFCxxx>
+<Tracking:>
+<  - https://github.com/chef/chef/issues/X>
+---
+
+# Title
+
+Define a governance system for transferring Supermarket namespace
+ownership.
+
+## Motivation
+
+    As a supermarket consumer,
+    I want to easily find well engineered, actively maintained
+    cookbooks, using meaningful names, so that I can be successful
+    with Chef.
+
+    As a cookbook author,
+    I want to publish cookbooks for technologies that occupy
+    abandoned, unmaintained, or otherwise forsaken namespace on the
+    supermarket.
+
+## Exploration
+
+As this point in time, this document serves as a place to collect
+thoughts and begin the discussion. This RFC is a true request for
+comments.
+
+It is clear that we need some sort of governance model to help things
+run more smoothly. The scope is the hard part. There are a ton of
+instances of abandoned cookbooks, namespace squatting, and low quality
+cookbooks occupying important names. People move on from the
+community, people die, change jobs, and simply lose interest. These
+needs to be a mechanism for dealing with this. Devising a fair,
+effective, and enforceable system is a huge undertaking.
+  
+Before proceeding, I would like to examine exactly what the
+Supermarket is. From there, we can decide the ends to which a
+governance model should strive to achieve, and start looking to other
+F/OSS communities for inspiration.
+
+The major conceptual models are:
+    
+- Chef as a language.
+- Chef as a software distribution.
+- Something in between
+   
+### Chef as a language
+    
+In this model, we draw parallels with programming languages. In most
+language communities, there is a system for contributing  and sharing
+libraries. Perl has CPAN. Ruby has Rubygems.org, Python has PyPI. Node 
+has NPM, and Java has Maven. Haskell has Hackage. Erlang has Hex.
+    
+There is a ton of momentum behind this mental model. Tooling such as
+Berkshelf was built with this in mind. Cookbooks are packages and the
+Supermarket is the hosting system.
+   
+Namespace governance is typically optimized for the preservation of
+backwards compatibility of individual components consumed by build
+systems.
+    
+When two libraries are concerned with the same topic, but work in
+different ways, they play semantic games to distinguish themselves.
+
+Some languages, by convention, use a special character for this
+purpose. 'alice/json' vs 'bob/json' is a popular method. Enabling this
```

Comment:
I like the idea of a namespace or package-like String that goes before a cookbook name.

> people could be told the top level /nginx that is stamped of approval by ops code is being changed and if they want to use the old one they can point to the namespace

I would strongly suggest we don't have a missing namespace mean that a cookbook's source could change. I would prefer we make namespaces required and assign some indicator value like 'none' to the existing cookbook names. All new cookbooks could use `none/foo` or `martinb3/foo`, but I don't think we should ever change `none/foo` to point to a different cookbook than `foo` does today. This would be nice and backwards compatible because we could always assume in any older code that a missing delimiter means `none/` for the namespace.